spongefish/src/transcript/mod.rs

```diff
@@ -0,0 +1,112 @@
+//! Abstract transcripts for interactive protocols.
+
+mod interaction;
+mod transcript_pattern;
+mod transcript_player;
+mod transcript_recorder;
+
+use core::error::Error;
+
+pub use self::{
+    interaction::{Interaction, InteractionHierarchy, InteractionKind, Length},
+    transcript_pattern::{TranscriptError, TranscriptPattern},
+    transcript_player::{InteractionError, TranscriptPlayer},
+    transcript_recorder::TranscriptRecorder,
+};
+
+/// Generic trait for transcripts that can process interactions.
+pub trait Transcript {
+    type Error: Error;
+
+    /// Process an interaction.
+    fn interact(&mut self, interaction: Interaction) -> Result<(), Self::Error>;
+}
+
+/// Extension trait for [`Transcript`].
+pub trait TranscriptExt {
```

Comment:
If this is an extension trait, shouldn't it look like this?

```suggestion
pub trait TranscriptExt: Transcript {
```